src/app/Mappings/components/MappingsTable.tsx

```diff
@@ -1,13 +1,119 @@
 import * as React from 'react';
-import { Mapping, MappingType, INetworkMapping, IStorageMapping } from '../types';
+import {
+  Mapping,
+  MappingType,
+  INetworkMapping,
+  IStorageMapping,
+  ICommonMapping,
+  INetworkMappingItem,
+  IStorageMappingItem,
+} from '../types';
+import { Level, LevelItem, Button, Pagination } from '@patternfly/react-core';
+import {
+  Table,
+  TableHeader,
+  TableBody,
+  ICell,
+  sortable,
+  compoundExpand,
+  classNames as classNamesTransform,
+  IRow,
+} from '@patternfly/react-table';
+import { useSortState, usePaginationState, useSelectionState } from '@app/common/hooks';
+import { IVMwareProvider } from '@app/Providers/types';
+import tableStyles from '@patternfly/react-styles/css/components/Table/table';
+import { OutlinedHddIcon, NetworkIcon, DatabaseIcon } from '@patternfly/react-icons';
+import ProviderStatus from '@app/Providers/components/ProvidersTable/ProviderStatus';
+import VMwareProviderActionsDropdown from '@app/Providers/components/ProvidersTable/VMware/VMwareProviderActionsDropdown';
+import VMwareProviderHostsTable from '@app/Providers/components/ProvidersTable/VMware/VMwareProviderHostsTable';
+import MappingsActionsDropdown from './MappingsActionsDropdown';
 
 interface IMappingsTableProps {
   mappings: Mapping[];
+  mappingType: string;
+  toggleAddEditModal: () => void;
 }
 
 const MappingsTable: React.FunctionComponent<IMappingsTableProps> = ({
   mappings,
+  mappingType,
+  toggleAddEditModal,
 }: IMappingsTableProps) => {
+  const getSortValues = (mapping: ICommonMapping) => {
+    const { name, sourceProvider, targetProvider } = mapping;
+    return [name, sourceProvider.name, targetProvider.name, ''];
+  };
+
+  const { sortBy, onSort, sortedItems } = useSortState(mappings, getSortValues);
+  const { currentPageItems, setPageNumber, paginationProps } = usePaginationState(sortedItems, 10);
+  React.useEffect(() => setPageNumber(1), [sortBy, setPageNumber]);
+
+  const {
+    selectedItems: expandedMappings,
+    toggleItemSelected: toggleMappingExpanded,
+  } = useSelectionState<ICommonMapping>(sortedItems);
```

Comment:
Same here, `Mapping` instead of `ICommonMapping`. Maybe I shouldn't have even exported it.